plutus-benchmark/bitwise/src/PlutusBenchmark/NQueens.hs

```diff
@@ -0,0 +1,77 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE MultiWayIf        #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+
+module PlutusBenchmark.NQueens (nqueens) where
+
+import PlutusTx.Builtins (replicateByte)
+import PlutusTx.Prelude
+
+-- Based on Qiu, Zongyan (February 2002). "Bit-vector encoding of n-queen problem". ACM SIGPLAN Notices. 37 (2): 68–70
+-- For simplicity, this only accepts multiples of 8 for the dimension (so 8, 16,
```

Comment:
It might be an interesting exercise to extend this to work for arbitrary n.  Let's not do that now though.